components/gitpod-db/src/typeorm/migration/1679312926829-OrganizationSlugIndex.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { indexExists } from "./helper/helper";
+
+const table = "d_b_team";
+const column = "slug";
+const orgIndex = "idx_slug";
+
+export class OrganizationSlugIndex1679312926829 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await indexExists(queryRunner, table, orgIndex))) {
+            await queryRunner.query(`CREATE INDEX \`${orgIndex}\` ON \`${table}\` (${column})`);
+        }
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {}
```

Comment:
Is an index considered distructive? Going back and forth should be safe when using `indexExists` here, no?
Anyways, adding the tear down step...

Reply:
While it's not destructive, it doesn't perform a full revert. What could happen is that we roll-back, the index stays in place and we choose to change strategy and say index multiple columns while re-using the same index name. In that case, the migration would fail to apply as the index name would've already existed.

I agree that in this case it's probably less likely, but generally getting in the habit of doing both up/down makes sense when we don't have as much access to the cell

Reply:
I agree this case is not super important. But also agree that adding rollbacks is becoming crucial with Dedicated, again. :100: